ci/run.sh

```diff
@@ -31,3 +40,14 @@ if [ -n "$COVERAGE" ]; then
     cd ..
     bash <(curl -s https://codecov.io/bash)
 fi
+
+# rebuild everything without ASAN and run it under valgrind
```

Comment:
I am against running with Valgrind on all the the different CI jobs. This will increase considerably the times of all of them (Specially since we are compiling and running test twice). What about just having 1 element in the Travis Matrix dedicated to the Valgrind checks?

Reply:
Sounds reasonable